File: build_panel/getting_rssds_and_certs/get_cert_rssd_map.R
library(xts)
library(readr)

quarters <- list.dirs("../../merged_data/", full.names = FALSE, recursive = FALSE)

fed_rssds_certs <- list()

for (q in quarters){
    tmp_data <- read_csv(paste0("../../merged_data/", q, "/bank_data.csv"))
    fed_rssds_certs[[q]] <- data.frame(cbind(tmp_data$cert, tmp_data$fed_rssd))
}

## not working-----
## Get all fed_rssds that have ever existed--------
all_fed_rssds_certs <- data.frame()

for(i in names(fed_rssds_certs)){rbind(fed_rssds_certs[i])}

unique(lapply(fed_rssds_certs, cbind))

saveRDS(all_fed_rssds_certs, "all_fed_rssds.rds")

File: build_panel/getting_rssds_and_certs/get_cert_rssd_map.py
# Get all fed_rssd and births/deaths

import os
import pandas as pd

quarters = os.listdir("./merged_data/") 

# convert to int to sort

quarters = [int(x) for x in quarters]

quarters.sort()

# create a structure to hold the fed_rssd, like R's list()

fed_rssd = {}

for i in quarters:
    fed_rssd[i] = pd.read_csv("./merged_data/" + str(i) + "/bank_data.csv")['fed_rssd']

# get rdds and cert map----not working

fed_rssd_and_cert = {} # is a dict

for i in quarters:
    fed_rssd_and_cert[i] = pd.read_csv("./merged_data/" + str(i) + "/bank_data.csv")[['cert', 'fed_rssd']]

all_rssd_and_cert = []

for i in quarters:
    all_rssd_and_cert.append(pd.DataFrame.from_dict(fed_rssd_and_cert.get(i)))


pd.DataFrame.to_csv(all_rssd_and_cert, "./all_rssd_and_cert.csv")
